app/models/observation_accuracy_experiment.rb

```diff
@@ -379,44 +409,33 @@ def generate_sample
     observations = Observation.includes( :taxon ).find( o )
 
     puts "Fetching continent groupings"
-    continent_obs = {}
     continents = Place.where( admin_level: -10 ).map( &:id )
-    o.each_slice( 100 ) do | batch |
-      oo = INatAPIService.observations( id: batch )
-      oo.results.each do | o_api |
-        continent = ( continents & o_api["place_ids"] ).first
-        continent_obs[o_api["id"]] = continent
-      end
-    end
+    continent_obs = get_continent_obs( o, continents )
     continent_key = Place.find( continents ).map {| a | [a.id, a.name] }.to_h
 
-    puts "Fetching number of descendants"
+    puts "Loading taxonomy"
+    load "lib/taxonomy_parser.rb"
```

Comment:
Best to use `require` instead of load for a production implementation. We could add the require to the top of the file `require "taxonomy_parser"`, or it might be better to include it in application.rb around where we make other custom requires https://github.com/inaturalist/inaturalist/blob/main/config/application.rb#L199. That'll make sure it gets loaded the same way as other libraries, and it'll be accessible throughout the codebase